dbms/src/Flash/Mpp/MPPTask.cpp

```diff
@@ -81,6 +81,44 @@ MPPTask::~MPPTask()
     LOG_FMT_DEBUG(log, "finish MPPTask: {}", id.toString());
 }
 
+void MPPTask::abortTunnels(const String & message, AbortType abort_type)
+{
+    switch (abort_type)
+    {
+    case AbortType::ONCANCELLATION:
+        closeAllTunnels(message);
+        break;
+    case AbortType::ONERROR:
+        RUNTIME_ASSERT(tunnel_set != nullptr, log, "mpp task without tunnel set");
+        tunnel_set->writeError(message);
+        break;
+    }
+}
+
+void MPPTask::abortReceivers()
+{
+    if (likely(receiver_set != nullptr))
+    {
+        receiver_set->cancel();
+    }
+}
+
+void MPPTask::abortDataStreams(AbortType abort_type)
+{
+    bool is_kill;
+    switch (abort_type)
+    {
+    case AbortType::ONCANCELLATION:
+        is_kill = true;
+        break;
+    case AbortType::ONERROR:
+        /// When abort type is ONERROR, it means MPPTask already known it meet error, so let the remaining task stop silently to avoid too many useless error message
+        is_kill = false;
+        break;
```

Comment:
Better to add default case, or use if/else instead of switch will make it more clear.

Reply:
Because there should be another AbortType `ONKILL`, I will use `if else` and change to `switch case` when `ONKILL` is added.